src/dc_simplify.rs

```diff
@@ -1,175 +1,169 @@
 use crate::dehtml::*;
 
-#[derive(Copy, Clone)]
-pub struct Simplify {
-    pub is_forwarded: bool,
-}
-
-/// Return index of footer line in vector of message lines, or vector length if
-/// no footer is found.
-///
-/// Also return whether not-standard (rfc3676, §4.3) footer is found.
-fn find_message_footer(lines: &[&str]) -> (usize, bool) {
+/// Remove standard (RFC 3676, §4.3) footer if it is found.
+fn remove_message_footer<'a>(lines: &'a [&str]) -> &'a [&'a str] {
     for (ix, &line) in lines.iter().enumerate() {
         // quoted-printable may encode `-- ` to `-- =20` which is converted
         // back to `--  `
```

Comment:
Is this comment actually done here?  Seems unrelated to the code so perhaps no longer relevant?

Reply:
According to the standard, we only have to check for "-- " (one space). The comment explains the reason why we check for "--  " (two spaces) too. I am not sure if it is still relevant because all the MIME parsing, encoding/decoding etc. is refactored.

Reply:
Oooh, i didn't realise the `=20` bit referred to the second space!  re-reading this now i see how this was saying the right thing all the time and i was just reading it wrong.  oh well